src/high-level/allocation.lisp

```diff
@@ -63,5 +83,13 @@ NOTE: Note that the finalizer may close over the allocated vector."
 (defun allocate (size &key element-type initial-element)
   "Allocate storage for a fresh tensor."
   (funcall *default-allocator* size
-           :element-type element-type
-           :initial-element initial-element))
+           element-type
+           initial-element))
+
+(defun finalize (x finalizer)
+  (when finalizer
+    #+allegro
```

Comment:
we might as well make this applicable to all lisps, since trusting execution of a finalizer is tenuous anyway